src/mainwindow.cpp

```diff
@@ -1660,15 +1661,20 @@ void MainWindow::on_action_Save_triggered() {
     updateMapList();
 }
 
-void MainWindow::on_tabWidget_2_currentChanged(int index)
+void MainWindow::on_mapViewTab_tabBarClicked(int index)
 {
+    int oldIndex = ui->mapViewTab->currentIndex();
+    if (index != oldIndex)
+        Scripting::cb_MapViewTabChanged(oldIndex, index);
```

Comment:
Same idea here--shouldn't the scripting hook be called after the actual index and switchover has completed?